src/MediaDeviceHandler.ts

```diff
@@ -68,18 +84,21 @@ export default class MediaDeviceHandler {
         setMatrixCallVideoInput(videoDeviceId);
     }
 
-    static setAudioOutput(deviceId: string) {
+    public setAudioOutput(deviceId: string) {
         SettingsStore.setValue("webrtc_audiooutput", null, SettingLevel.DEVICE, deviceId);
+        this.emit(MediaDeviceHandlerEvent.AudioOutputChanged, deviceId);
     }
 
-    static setAudioInput(deviceId: string) {
+    public setAudioInput(deviceId: string) {
         SettingsStore.setValue("webrtc_audioinput", null, SettingLevel.DEVICE, deviceId);
         setMatrixCallAudioInput(deviceId);
+        this.emit(MediaDeviceHandlerEvent.AudioInputChanged, deviceId);
     }
 
-    static setVideoInput(deviceId: string) {
+    public setVideoInput(deviceId: string) {
         SettingsStore.setValue("webrtc_videoinput", null, SettingLevel.DEVICE, deviceId);
         setMatrixCallVideoInput(deviceId);
+        this.emit(MediaDeviceHandlerEvent.VideoInputChanged, deviceId);
```

Comment:
Will these work mid-call at this point or not?

Reply:
No, that would require, js-sdk changes. I hope to do that at some point too, though

Reply:
That's what I thought, can we get comments stating that at the very least? It seems weird it emits an event that we know will leave the app in a split-brained state if it was actually listening on it